exporter/prometheusexporter/README.md

```diff
@@ -19,6 +19,11 @@ The following settings can be optionally configured:
 - `metric_expiration` (default = `5m`): defines how long metrics are exposed without updates
 - `resource_to_telemetry_conversion`
   - `enabled` (default = false): If `enabled` is `true`, all the resource attributes will be converted to metric labels by default.
+- `prometheus_convention` (default = false): If `true`, the OpenTelemetry
```

Comment:
I agree with you, I'm just concerned with the potential problems this may cause. How do you see this feature-gate working? I'm confused on how that would work.

Also, important question: Are we trying to comply with [Prometheus' metric naming convention](https://prometheus.io/docs/practices/naming/) or with [OpenMetrics](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md)? OpenMetrics has more guidelines (like appending `_sum` for *Summary* metrics, `_gcount` and `_gsum` for *GaugeHistogram*, etc.)

Reply:
See [this](https://github.com/open-telemetry/opentelemetry-collector/tree/e8dd9536f5e53228606c325f12d4bfc4b4293acc/service/featuregate#collector-feature-gates) for the overview of feature-gates.  They are a CLI flag which can enable/disable functionality, and it enables progression for a feature from Alpha (off by default; can be enabled) to Beta (on by default, can be disabled) to GA (on by default, cannot be disabled).  Feature-gates should be used in cases where long-term configuration is not needed, but a more "careful" transition from one state to another is needed.

We start with the feature-gate disabled by default, while we validate that this is a change we _actually_ want to make, and make sure other pieces of functionality (e.g. changes in the prometheus receiver) are in-place.  Then we graduate the feature-gate(s) to beta, and get user-feedback.  If users encounter issues, they can always disable the functionality.

Technically, we should be complying with prometheus' conventions, since [we don't](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1274b14412ede4725184ad17dc8f90f0473f79ea/exporter/prometheusexporter/prometheus.go#L58) [enable openmetrics](https://github.com/prometheus/client_golang/blob/1f81b3e9130fe21c55b7db82ecb2d61477358d61/prometheus/promhttp/http.go#L156) on the prometheus handler.  However, note that suffixing based on openmetrics vs prometheus text format is handled by the prometheus client ([example where openmetrics appends _total](https://github.com/prometheus/common/blob/main/expfmt/openmetrics_create.go#L90)).  So as long as we use the prometheus client, we adding type suffixes correctly.  Units, on the other hand, we need to do ourselves.